### PR TITLE
feat(Breadcrumbs): add `renderItem` property

### PR DESCRIPTION
**Pull request by GermanVor**
https://github.com/gravity-ui/uikit/issues/1287

### CODE REVIEW

**Review by korvin89 — CHANGES_REQUESTED** [review]

File: src/components/Breadcrumbs/BreadcrumbsItem.tsx [hunk 1]
```diff
@@ -3,56 +3,67 @@ import React from 'react';
...
     renderItem,
 }: Props<T>) {
-    const itemTitle = data.title || data.text;
+    const children = renderItemContent
+        ? renderItemContent(item, isCurrent, isPrevCurrent)
+        : item.text;
 
-    const item = renderItem ? renderItem(data, isCurrent, isPrevCurrent) : data.text;
+    if (renderItem) {
+        return renderItem({item, children, isCurrent, isPrevCurrent});
```
[the remark this answers is not in the document]
Comment (GermanVor) [reply src/components/Breadcrumbs/BreadcrumbsItem.tsx:36]: Why it is breaking changes ?

There is no exports for this Item  and if user imported it like this - `@gravity-ui/uikit/build/cjs/components/Breadcrumbs/BreadcrumbsItem` - Ts and eslint will inform that prev `renderItem` is not the same as new one.

File: src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 2]
```diff
@@ -30,12 +31,33 @@ type ButtonBreadcrumbsItem = {
 
 export type BreadcrumbsItem = LinkBreadcrumbsItem | ButtonBreadcrumbsItem;
 
+export type RenderItemContent<T extends BreadcrumbsItem> = (
```
[the remark this answers is not in the document]
Comment (GermanVor) [reply src/components/Breadcrumbs/Breadcrumbs.tsx]: 0137809f0a74902c20e0aba833c52e10e7cdd5a6

File: src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 3]
```diff
@@ -30,12 +31,33 @@ type ButtonBreadcrumbsItem = {
...
+) => React.ReactNode;
+
+export type RenderItemProps<T extends BreadcrumbsItem = BreadcrumbsItem> = {
+    children: React.ReactNode;
+    item: T;
+    isCurrent: boolean;
+    isPrevCurrent: boolean;
+};
+
+export type RenderItem<T extends BreadcrumbsItem> = (props: RenderItemProps<T>) => React.ReactNode;
```
[the remark this answers is not in the document]
Comment (GermanVor) [reply src/components/Breadcrumbs/Breadcrumbs.tsx]: 0137809f0a74902c20e0aba833c52e10e7cdd5a6

File: src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 4]
```diff
@@ -30,12 +31,33 @@ type ButtonBreadcrumbsItem = {
...
+    isCurrent: boolean,
+    isPrevCurrent: boolean,
+) => React.ReactNode;
+
+export type RenderRootContent<T extends BreadcrumbsItem> = (
+    item: T,
+    isCurrent: boolean,
+) => React.ReactNode;
+
+export type RenderItemProps<T extends BreadcrumbsItem = BreadcrumbsItem> = {
```
[the remark this answers is not in the document]
Comment (GermanVor) [reply src/components/Breadcrumbs/Breadcrumbs.tsx]: 0137809f0a74902c20e0aba833c52e10e7cdd5a6

**Review by korvin89 — APPROVED** [review]

**Pull request merged**